api/src/DataPersister/InvitationDataPersister.php

```diff
@@ -2,48 +2,72 @@
+        $rejectListener = CustomActionListener::of(Invitation::REJECT, fn ($data) => $this->onReject($data));
+        parent::__construct(
+            Invitation::class,
+            $dataPersisterObservable,
+            customActionListeners: [$acceptListener, $rejectListener]
+        );
     }
 
+    /**
+     * Override supports to omit check if Invitation is a managed Entity (which it's not).
+     */
     public function supports($data, array $context = []): bool {
         return $data instanceof Invitation;
     }
 
+    public function onAccept($data): CampCollaboration {
+        $campCollaboration = $this->campCollaborationRepository->findByInviteKey($data->inviteKey);
+        $campCollaboration->user = $this->security->getUser();
+        $campCollaboration->status = CampCollaboration::STATUS_ESTABLISHED;
+        $campCollaboration->inviteKey = null;
+        $campCollaboration->inviteEmail = null;
+
+        return $campCollaboration;
+    }
+
+    public function onReject($data): CampCollaboration {
+        $campCollaboration = $this->campCollaborationRepository->findByInviteKey($data->inviteKey);
+        $campCollaboration->user = $this->security->getUser();
+        $campCollaboration->status = CampCollaboration::STATUS_INACTIVE;
+        $campCollaboration->inviteKey = null;
+
+        return $campCollaboration;
+    }
+
     /**
-     * @param Invitation $data
+     * Override persist method to return Invitation instead of the persisted CampCollaboration.
      */
     public function persist($data, array $context = []) {
-        if (Invitation::ACCEPT === ($context['item_operation_name'] ?? null)) {
-            $campCollaboration = $this->campCollaborationRepository->findByInviteKey($data->inviteKey);
-            $campCollaboration->user = $this->security->getUser();
-            $campCollaboration->status = CampCollaboration::STATUS_ESTABLISHED;
-            $campCollaboration->inviteKey = null;
-            $campCollaboration->inviteEmail = null;
-            //TODO: add MaterialList for User
-            $this->dataPersister->persist($campCollaboration);
-
-            $data->userAlreadyInCamp = true;
-        } elseif (Invitation::REJECT === ($context['item_operation_name'] ?? null)) {
-            $campCollaboration = $this->campCollaborationRepository->findByInviteKey($data->inviteKey);
-            $campCollaboration->user = $this->security->getUser();
-            $campCollaboration->status = CampCollaboration::STATUS_INACTIVE;
-            $campCollaboration->inviteKey = null;
-            $this->dataPersister->persist($campCollaboration);
-        }
+        parent::persist($data, $context);
+
+        return $data;
     }
 
-    public function remove($data, array $context = []) {
+    /**
+     * {@inheritDoc}
+     */
+    public function beforeRemove($data): ?BaseEntity {
         throw new \RuntimeException('remove is not supported for '.Invitation::class);
     }
```

Comment:
Ist das nicht toter Code wenn wir die Aktion in der `ApiResource` nicht aktivieren und einen Test dafür haben, dass 405 Method Not Supported kommt?

Reply:
ContextAwareDataPersisterInterface verlangt, dass die methode remove überschrieben wird.
Der default use case ist wahrscheinlich, die Entity zu löschen, das ist jetzt in AbstractDataPersister drin. (Anders als zuvor.)

Ich finde deinen Vorschlag gut. Es bildet das gleiche verhalten ab, wie wenn wir keinen DataPersister hätten.